### ISSUE TITLE
SDL1.2 : SDL_AddTimer not working

### ISSUE THREAD

**Issue by vichoca9** [pspdev/psp-packages#184]
I was compiling POWDER for PSP again based on the files provided on its latest build, with current toolchain from  pspdev's page and with `pspdev sdl 1.2.15-5 [installed]` latest sdl.

For some reason, the timer thread doesnt do anything, yet the game works fine on Linux SDL with almost no changes, so the Callback function (or the thread) might be not getting called.

I might update this once I test the threads issue more deeply, maybe with a simpler app.

**Comment by sharkwouter** [comment]
Interesting, the code for the SDL 1.2 port which we host can be found here: https://github.com/pspdev/SDL/tree/psp-v.1.2.15

It has been a while since anyone touched it.

**Comment by vichoca9** [comment]
Nope, the schrodinger bug keeps happening. I even updated the game to SDL2 (and now it works), but SDL1.2 still doesnt. It seems that SDL_PumpEvents or SDL_WaitEvents might be involved, I will keep this open JUST in case someone else has this bug.

**Comment by sharkwouter** [comment]
Do you have a code example that shows the bug by any chance?

**Comment by vichoca9** [comment]
SDL1.2 testtimer.c works in Linux but not on PSP. I provided a makefile (run `make all` for psp, else compile normally with command on PC) and the modified test.
[testtimer.zip](https://github.com/user-attachments/files/16734849/testtimer.zip)

**Comment by vichoca9** [comment]
SDL2 version runs as expected (modified from the same test!):
[testtimer_SDL2.zip](https://github.com/user-attachments/files/16734912/testtimer_SDL2.zip)

**Comment by sharkwouter** [comment]
Thank for the example. I'm glad it works on SDL2 at least. That probably means it works on SDL3 as well. My guess right now is that this fix never arrived in SDL 1.2: https://github.com/libsdl-org/SDL/pull/9340

But there could also be other causes. SDL 1.2 is not so commonly used and there have been some changes in the API in the SDK.
